src/scrot.c

```diff
@@ -590,20 +590,20 @@ static Bool scrotXEventVisibility(Display *dpy, XEvent *ev, XPointer arg)
 static char *imPrintf(char *str, struct tm *tm, char *filenameIM,
     char *filenameThumb, Imlib_Image im)
 {
-    char *c;
     char buf[20];
     Stream ret = {0};
     long hostNameMax = 0;
     char strf[4096];
     char *tmp;
     struct stat st;
 
-    if (strftime(strf, 4095, str, tm) == 0)
+    const size_t strfLen = strftime(strf, sizeof(strf), str, tm);
```

Comment:
note: Replaced the hardcoded 4095 with `sizeof(strf)`.

Previously it was one less than sizeof(strf), but there was real reason for that. `strftime` will return 0 if the resulting string (including the nul-byte) was too big.